src/main/scala/dxWDL/compiler/DxNI.scala

```diff
@@ -143,18 +143,39 @@ case class DxNI(verbose: Verbose, language: Language.Value) {
   }
 
   private def documentFromTasks(tasks: Vector[TAT.Task]): TAT.Document = {
+    def createDocument(docTasks: Vector[TAT.Task]): TAT.Document = {
+      TAT.Document(
+          None,
+          "",
+          TAT.Version(codeGen.wdlVersion, TextSource.empty),
+          docTasks,
+          None,
+          TextSource.empty,
+          CommentMap.empty
+      )
+    }
+
     // uniquify and sort tasks
     val sortedUniqueTasks =
       tasks.map(t => t.name -> t).toMap.values.toVector.sortWith(_.name < _.name)
-    TAT.Document(
-        None,
-        null,
-        TAT.Version(codeGen.wdlVersion, null),
-        sortedUniqueTasks,
-        None,
-        null,
-        CommentMap.empty
-    )
+    // validate each task and warn if it doesn't generate valid WDL
+    val parser = ParseWomSourceFile(verbose.on)
+    val validTasks = sortedUniqueTasks.flatMap { task =>
+      try {
+        // TODO: currently we always generate WDL 1.0 - other versions of the code generator
+        //  need to be implemented in wdlTools
```

Comment:
Is there a way to log the version of WDL we generate?

Reply:
Well, it is present in the version statement in the WDL file. And if there is no version statement, it is assumed to be draft-2.

Reply:
Right! OK, if the tests pass that, LGTM, thanks!